File: viewer/src/main/java/com/tianxing/hotflyer/viewer/fragment/favourite/FavouriteMovieFragment.java
package com.tianxing.hotflyer.viewer.fragment.favourite;

import android.support.v7.widget.RecyclerView;

import com.tianxing.hotflyer.viewer.JAViewer;
import com.tianxing.hotflyer.viewer.adapter.ItemAdapter;
import com.tianxing.hotflyer.viewer.adapter.MovieAdapter;
import com.tianxing.hotflyer.viewer.view.decoration.MovieItemDecoration;

/**
 * Project: JAViewer
 */

public class FavouriteMovieFragment extends FavouriteFragment {
    @Override
    public ItemAdapter adapter() {
        return new MovieAdapter(JAViewer.CONFIGURATIONS.getStarredMovies(), this.getActivity());
    }

    @Override
    public RecyclerView.ItemDecoration decoration() {
        return new MovieItemDecoration();
    }
}

File: viewer/src/main/java/com/tianxing/hotflyer/viewer/fragment/favourite/FavouriteMovieFragment.kt
package com.tianxing.hotflyer.viewer.fragment.favourite

import android.support.v7.widget.RecyclerView

import com.tianxing.hotflyer.viewer.JAViewer
import com.tianxing.hotflyer.viewer.adapter.ItemAdapter
import com.tianxing.hotflyer.viewer.adapter.MovieAdapter
import com.tianxing.hotflyer.viewer.view.decoration.MovieItemDecoration

/**
 * Project: JAViewer
 */

class FavouriteMovieFragment : FavouriteFragment() {
    override fun adapter(): ItemAdapter<*, *> {
        return MovieAdapter(JAViewer.CONFIGURATIONS!!.starredMovies, this.activity)
    }

    override fun decoration(): RecyclerView.ItemDecoration? {
        return MovieItemDecoration()
    }
}
